incubator/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/stream/GrpcServerFactory.java

```diff
@@ -0,0 +1,1369 @@
+        final long affinity = begin.affinity();
+        final long initialId = begin.streamId();
+        final long sequence = begin.sequence();
+        final long acknowledge = begin.acknowledge();
+        final OctetsFW extension = begin.extension();
+        final HttpBeginExFW httpBeginEx = extension.get(httpBeginExRO::tryWrap);
+
+        MessageConsumer newStream = null;
+
+        if (!isGrpcRequestMethod(httpBeginEx))
+        {
+            doHttpResponse(network, traceId, authorization, affinity, routeId, initialId, sequence, acknowledge,
+                HEADER_VALUE_STATUS_405, HEADER_VALUE_GRPC_INTERNAL_ERROR);
+            newStream = (t, b, i, l) -> {};
+        }
+        else
+        {
+            final GrpcBindingConfig binding = bindings.get(routeId);
+
+            final GrpcMethodConfig methodConfig = binding.resolveGrpcMethodConfig(httpBeginEx);
+
+            if (methodConfig != null)
+            {
+                newStream = newInitialGrpcStream(begin, network, httpBeginEx, methodConfig);
+            }
+            else
+            {
+                doHttpResponse(network, traceId, authorization, affinity, routeId, initialId, sequence, acknowledge,
+                    HEADER_VALUE_STATUS_200, HEADER_VALUE_GRPC_UNIMPLEMENTED);
+                newStream = (t, b, i, l) -> {};
+            }
+        }
+
+        return newStream;
+    }
+
+    public MessageConsumer  newInitialGrpcStream(
+        final BeginFW begin,
+        final MessageConsumer network,
+        final HttpBeginExFW httpBeginEx,
+        final GrpcMethodConfig methodConfig)
+    {
+        final long routeId = begin.routeId();
+        final long initialId = begin.streamId();
+        final long replyId = supplyReplyId.applyAsLong(initialId);
+        final long affinity = begin.affinity();
+        final long traceId = begin.traceId();
+        final long authorization = begin.authorization();
+        final long sequence = begin.sequence();
+        final long acknowledge = begin.acknowledge();
+
+        GrpcBindingConfig binding = bindings.get(routeId);
+
+        GrpcRouteResolver route = null;
+
+        if (binding != null)
+        {
+            route = binding.resolve(begin.authorization(), httpBeginEx, methodConfig);
+        }
+
+        MessageConsumer newStream = null;
+
+        if (route != null)
+        {
+            final ContentType contentType = asContentType(route.contentType);
+
+            newStream = new GrpcServer(
+                network,
+                routeId,
+                initialId,
+                replyId,
+                affinity,
+                route.id,
+                contentType,
+                route.metadata,
+                methodConfig)::onNetMessage;
+        }
+        else
+        {
+            doHttpResponse(network, traceId, authorization, affinity, routeId, initialId, sequence, acknowledge,
+                HEADER_VALUE_STATUS_200, HEADER_VALUE_GRPC_UNIMPLEMENTED);
```

Comment:
Is it really http status 200 for a grpc error like this?

Reply:
Yeah it is 200